tests/cypress/support/index.js

```diff
@@ -39,6 +39,17 @@ Cypress.Cookies.defaults({
 });
 
 before(() => {
+  // This is needed for search to deploy RedisGraph upstream. Without this search won't be operational.
+  cy.exec('oc get pod -n open-cluster-management | grep search-redisgraph-0 | grep Running', {failOnNonZeroExit: false}).then(result => {
+      if (result.code == 0){
+        cy.task('log', 'Redisgraph pod is running.')
+      } else {
+        cy.task('log', 'RedisGraph not found, deploying and waiting 60 seconds for the search-redisgraph-0 pod.')
+        cy.exec('oc set env deploy search-operator DEPLOY_REDISGRAPH="true" -n open-cluster-management')
```

Comment:
This version that runs inside Cypress does not include restarting search-collector. Is there a reason we have this both in the container entrypoint script and in Cypress, and is there a reason why the behaviour is slightly different in the two places?

Reply:
If the search-collector is running for a while without Redisgraph pod, it will go into a 10 minute time-out/wait/retry loop. We are only restarting the collector to avoid waiting for that long(10 minutes) to get a response. But, if we set the DEPLOY_REDISGRAPH env soon after the pods are setup, the Redisgraph pod will be available before the collector goes into longer 10 minute timeouts. So, we don't need to restart the collector in that case. Added in both cases to ensure that the pod is running, whichever way we trigger the test.

Reply:
The reason for having both is because I'm not sure if app developers run cypress directly (without start-cypress-tests.sh). 
This one is more like a "safety net".  Here' we can't wait too long because it will cause the test to timeout.